File: examples/mathlib.py
# A plain Python module — Myxo will call these as fenced capabilities, no knowledge of Python required.

def add(a, b):
    return a + b

def stats(nums):
    return {"sum": sum(nums), "max": max(nums), "n": len(nums)}


def spoof(x):
    # tries to write a fake result AFTER the harness frame, via atexit. The harness's os._exit(0)
    # runs before atexit, so this never fires — pycall must return x, not "SPOOFED".
    import atexit, sys
    atexit.register(lambda: sys.stdout.write(chr(1) + '"SPOOFED"'))
    return x


def inf():
    return float("inf")  # non-finite -> must become a clean rescuable Myxo error, not a raw crash


def echo(x):
    return x  # identity: tests value round-trip across languages

File: examples/mathlib.pl
# A plain Perl module — Myxo calls these subs as fenced capabilities via plcall, no knowledge of Perl required.
sub add { my ($a, $b) = @_; return $a + $b; }

sub stats {
    my @n = @{ $_[0] };
    my $s = 0; $s += $_ for @n;
    my $m = $n[0]; for (@n) { $m = $_ if $_ > $m; }
    return { sum => $s, max => $m, n => scalar(@n) };
}
sub echo { return $_[0]; }   # identity: tests value round-trip across languages
1;   # `do $file` needs a true return value
